### PR TITLE
Fix: allow zero-fee transactions in LowestFee metric

### PR DESCRIPTION
**Pull request by evanlinjin**
Change assertion from fee > 0 to fee >= 0 to handle valid zero-fee transactions. Add test case to verify the metric works correctly when target fee rate is zero.

Also make clippy happy and upgrade test dependencies.

### CODE REVIEW

File: tests/lowest_fee.rs [hunk 1]
```diff
@@ -267,3 +267,55 @@ fn adding_another_input_to_remove_change() {
     assert!(score <= best_solution_score);
     assert_eq!(cs.selected_indices(), best_solution.selected_indices());
 }
+
+#[test]
+fn zero_fee_tx() {
```
Comment (jp1ac4) [inline tests/lowest_fee.rs:272]: Not sure if it matters, but AFAICT this test produces a solution with a non-zero fee (130 sats).

Comment (evanlinjin) [reply tests/lowest_fee.rs:272]: If it's a solution with change then it's a problem. If it's without change, it's expected as we have a long-term feerate so that greatly disincentives the solution with change.

Comment (evanlinjin) [reply tests/lowest_fee.rs:272]: But maybe we should check that a solution with change has 0 fee.

Comment (jp1ac4) [reply tests/lowest_fee.rs:272]: Ah OK, the solution doesn't have change so that's fine. I wasn't sure if the intention of the test was to produce a transaction with zero fee.
